core/internal/api/ratelimittracker_test.go

```diff
@@ -0,0 +1,158 @@
+package api_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/wandb/wandb/core/internal/api"
+)
+
+func TestInitialRateLimit(t *testing.T) {
+	rl := api.NewRateLimitTracker(api.RateLimitTrackerParams{
+		MinPerSecond: 1,
+		MaxPerSecond: 10,
+	})
+
+	assert.EqualValues(t, 10, rl.TargetRateLimit())
+}
+
+func TestUpdate_NoSmoothing_FindsOptimalRate(t *testing.T) {
+	rl := api.NewRateLimitTracker(api.RateLimitTrackerParams{
+		MinPerSecond:           0.01,
+		MaxPerSecond:           10,
+		Smoothing:              0,
+		MinRequestsForEstimate: 1,
+	})
+
+	rl.UpdateEstimates(time.Time{}, 20, 8)
```

Comment:
nit: can you add comments on the arguments for readability?

Reply:
honestly, it is really hard for me to understand the math and i tried reading the documentation on the implementation. i think a more detailed explanation in one place (probably on the relevant top method) would be really helpful.